ivy/stateful/module.py

```diff
@@ -832,6 +830,27 @@ def train(self, mode: bool = True):
             if isinstance(module, ivy.Module):
                 module.train(mode=mode)
 
+    def to_device(self, device):
+        # moves the weights and buffers
+        # to the specified device
+
+        # moving weights and buffers to new device
+        for key, obj in self.v.items():
+            if (
+                isinstance(obj, ivy.Module)
+                or ivy.is_ivy_array(obj)
+                or isinstance(obj, ivy.Container)
+            ):
+                obj.to_device(device)
+            else:
+                ivy.to_device(obj, device=device)
```

Comment:
Actually, `to_device` isn't inplace by default, so you'd need to either do a `ivy.to_device(obj, device=device, out=obj)` or a `obj._data = ivy.to_device(obj).data`